src/Widgets/LidCloseActionComboBox.vala

```diff
@@ -40,15 +42,44 @@ namespace Power {
 
             hexpand = true;
             update_current_action ();
+            previous_active = active;
             changed.connect (on_changed);
         }
 
+        private bool set_active_with_permission (int index_) {
+            // Returns true on success
+
+            var permission = MainView.get_permission ();
```

Comment:
This may return `null` so requires a check before using `permission`

Reply:
We should probably display an error message or something. Otherwise combobox will just reset to a previous item and user will be confused

Reply:
Yes, ideally, but I do not know how likely it is that this situation will occur - hopefully it is rare.  I think further change to the UI, if required, can be left for the `gtk4` branch.